pkgs/games/etlegacy/default.nix

```diff
@@ -0,0 +1,124 @@
+{ stdenv
+, lib
+, makeWrapper
+, writeScriptBin
+, fetchgit
+, fetchFromGitHub
+, fetchzip
+, fetchurl
+, runCommand
+, cmake
+, git
+, glew
+, SDL2
+, zlib
+, minizip
+, libjpeg
+, curl
+, lua5_4
+, libogg
+, libtheora
+, freetype
+, libpng
+, sqlite
+, openal
+, unzip
+, cjson
+}:
+
+let
+  version = "2.81.1";
+  pkgname = "etlegacy";
+  lua = lua5_4;
+  mirror = "https://mirror.etlegacy.com";
+  fetchAsset = {asset, sha256}: fetchurl
+    { url = mirror + "/etmain/" + asset;
+    inherit sha256;
+    };
+  pak0 = fetchAsset
+    { asset = "pak0.pk3";
+    sha256 = "712966b20e06523fe81419516500e499c86b2b4fec823856ddbd333fcb3d26e5";
+    };
+  pak1 = fetchAsset
+    { asset = "pak1.pk3";
+    sha256 = "5610fd749024405b4425a7ce6397e58187b941d22092ef11d4844b427df53e5d";
+    };
+  pak2 = fetchAsset
+    { asset = "pak2.pk3";
+    sha256 = "a48ab749a1a12ab4d9137286b1f23d642c29da59845b2bafc8f64e052cf06f3e";
+    };
+  fakeGit = writeScriptBin "git" ''
+    #! ${stdenv.shell} -e
+    if [ "$1" = "describe" ]; then
+      if [ -r VERSION.txt ]; then
+        . <(sed 's/^\(VERSION_[A-Z]\+\) \(.*\)/\1=\2/' VERSION.txt)
+        VERSION="$VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH"
+      fi
+      if [ "$VERSION" != "" ]; then
+        echo $VERSION
+      else
+        echo "Unable to determine version"
+        exit 1
+      fi
+    fi
+  '';
+  mainProgram = if stdenv.hostPlatform.system == "i686-linux" then "etl.i386" else "etl.x86_64";
+in stdenv.mkDerivation rec {
+  pname = pkgname;
+  inherit version;
+
+  src = fetchgit {
+    url = "https://github.com/etlegacy/etlegacy.git";
+    rev = "refs/tags/v" + version;
+    sha256 = "sha256-CGXtc51vaId/SHbD34ZeT0gPsrl7p2DEw/Kp+GBZIaA=";  # 2.81.1
+    fetchSubmodules= false;
+  };
+
+  nativeBuildInputs = [ cmake fakeGit git makeWrapper unzip cjson ];
+  buildInputs = [
+    glew SDL2 zlib minizip libjpeg curl lua libogg libtheora freetype libpng sqlite openal
+  ];
+
+  preBuild = ''
+    # Required for build time to not be in 1980
+    export SOURCE_DATE_EPOCH=$(date +%s)
+    # This indicates the build was by a CI pipeline and prevents the resource
+    # files from being flagged as 'dirty' due to potentially being custom built.
+    export CI="true"
```

Comment:
are these really needed ? this looks a bit weird and might be worth doing sthg upstream. Now I dont want to delay the merge of this further, I know it's already quite some work so if you are not interested let's merge as is.

Reply:
Those were supplied by one of the maintainers of the upstream build, because without `CI=true` the game version displays as e.g. `2.81-dirty` on the main menu and elsewhere in the game, which indicates a non-release build.
The build time is less prominent but I think by convention the upstream project likes it to be set as at the time the build happened.

Reply:
ok let's say I disagree with their semantic but I am bikeshedding, let's not care about that :) thanks